ci-operator/step-registry/openshift/e2e/test/openshift-e2e-test-commands.sh

```diff
@@ -151,28 +151,34 @@ if [[ "${CLUSTER_TYPE}" == gcp ]]; then
 fi
 
 function upgrade() {
-    set -x
-    TARGET_RELEASES="${OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE:-}"
+    set -x &&
+    TARGET_RELEASES="${OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE:-}" &&
     if [[ -f "${SHARED_DIR}/override-upgrade" ]]; then
-        TARGET_RELEASES="$(< "${SHARED_DIR}/override-upgrade")"
+        TARGET_RELEASES="$(< "${SHARED_DIR}/override-upgrade")" &&
         echo "Overriding upgrade target to ${TARGET_RELEASES}"
-    fi
+    fi &&
     openshift-tests run-upgrade "${TEST_UPGRADE_SUITE}" \
         --to-image "${TARGET_RELEASES}" \
         --options "${TEST_UPGRADE_OPTIONS-}" \
         --provider "${TEST_PROVIDER}" \
         -o "${ARTIFACT_DIR}/e2e.log" \
         --junit-dir "${ARTIFACT_DIR}/junit" &
-    wait "$!"
+    wait "$!" &&
     set +x
```

Comment:
I'm not sure all these `&&`'s are required, can't you just save the exit code of openshift-tests and return it at the end? 

The problem is the $? we're saving right now is the return code of `set +x`, which is always successful.

Reply:
We discussed on Slack.  My suggestion only works for the simplest use case, `upgrade_conformance` invokes many commands.  We'd have to save, check, set +x, and return on error for each one -- which is probably equal ugliness to chained &'s.

Reply:
Using `&&` chaining is saving the exit code (and skipping subsequent commands within the function) on error.  You could:

```sh
local exit_code=0
bla
blah
openshift-tests ... &
wait "$!" || exit_code="$?"
set +x
return "$exit_code"
```

But in that case, the caller would still receive a 0 exit code from the function call in cases where `bla`, `blah`, or `set` happened to fail.  In this particular case, there isn't all that much else going on that's likely to fail, but `&&` chaining the whole function doesn't take many characters, and saves folks from the trouble of having to think through whether the other commands are likely to fail in ways that would be troublesome to ignore, by avoiding any chance of ignored error.